File: src/Application/KB_KeyApp.cpp
#include "KB_KeyApp.h"
KB_KeyApp::KB_KeyApp() {}
void KB_KeyApp::collectSignal() {}
void KB_KeyApp::reloadKeyLayer(int layer)
{
    currentLayer = layer;
    cout << "currentLayer:" << currentLayer;
    // TODO-加载keymap配置
}
void KB_KeyApp::parseKeyState() {}
void KB_KeyApp::pressKeys() {}

File: python/prototype/KB_KeyApp.py
# -*- coding: utf-8 -*-
# @Time    : 2021/4/11 7:36 PM
# @Author  : Patrick(贤人)
# @Email   : patrick.zxx@alibaba-inc.com
# @File    : KB_KeyApp.py
# @Software: ArduinoKeyboard
# @Description: KB_KeyApp

import random
from python.prototype import keyconfig


class KB_KeyApp():
    def initLayer(self, layer):
        self.c_layer = layer
        self.keyState = [0 for i in range(keyconfig.NUM_KEYS)]

    def collectSignal(self):
        for i in range(random.randint(3)):
            rnd_id = random.randint(keyconfig.NUM_KEYS)
            self.keyState[rnd_id] = 1 - self.keyState[rnd_id]
        return self.keyState

    def parseKeyState(self):
        pass

    def pressKeys(self):
        pass


if __name__ == "__main__":
    pass
